### PR TITLE
GeoJSON export polygon detection + preset injection

### PR DESCRIPTION
**Pull request by gmaclennan**
- Uses iD’s polygon detection algorithm based on custom presets.
- Injects preset icons and ids into exported geojson
- Replaces ‘:’ with ‘_’ for legacy data/tag names

### CODE REVIEW

File: lib/preset-matcher.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+      for (var k in preset.tags) {
+        k = k.replace(':', '_')
+        ;(g[k] = g[k] || []).push(preset)
+      }
+    }
+  }
+
+  return match
+
+  function match (feature) {
```
Comment (noffle) [inline lib/preset-matcher.js]: Nit: if you made `index` a parameter, you could break this function out of the closure and make it pure.

File: lib/preset-matcher.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+          match = keyMatches[i]
+        }
+      }
+    }
+
+    return match || null
+  }
+}
+
+function matchScore (preset, props) {
```
Comment (noffle) [inline lib/preset-matcher.js]: Comment idea: what does 'score' mean in this context?

File: lib/preset-matcher.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
+module.exports = function presetMatcher (presets) {
```
Comment (noffle) [inline lib/preset-matcher.js]: Bonus points for a comment describing what this inner module does.

File: server.js [hunk 4]
```diff
@@ -4,7 +4,7 @@ var sneakernet = require('hyperlog-sneakernet-replicator')
 var net = require('net')
 
 var body = require('body/any')
-var qs = require('querystring')
+var parseUrl = require('url').parse
```
Comment (noffle) [inline server.js]: This was the only use site of `querystring`; you can drop it from `package.json`.

File: server.js [hunk 5]
```diff
@@ -16,10 +16,29 @@ var randombytes = require('randombytes')
...
 var metadata = userConfig.getSettings('metadata')
+var presets = userConfig.getSettings('presets')
+
+var matchPreset = require('./lib/preset-matcher')(presets.presets)
+var isPolygonFeature = require('./lib/polygon-feature')(presets.presets)
 
 var Bonjour = require('bonjour')
 var HTTP_PORT = 3198
 
+var featureMap = function (f) {
```
Comment (noffle) [inline server.js]: Nit: switch to hoisting-style function decl and move to bottom of file.

File: server.js [hunk 6]
```diff
@@ -31,6 +50,11 @@ module.exports = function (osm) {
 
   var server = http.createServer(function (req, res) {
     console.log(req.method, req.url)
+    var params = parseUrl(req.url, true).query
+    var bbox = [params.minlon, params.minlat, params.maxlon, params.maxlat]
```
Comment (noffle) [inline server.js]: Did you mean to define `bbox` inside of the handler for the export route?

File: lib/polygon-feature.js [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+
+    var value = d.tags[key]
+    if (d.geometry.indexOf('area') === -1 &&
+                d.geometry.indexOf('line') !== -1 &&
+                key in areaKeys && value !== '*') {
+      areaKeys[key][value] = true
+    }
+  })
+
+  return function isPolygon (coords, tags) {
```
Comment (gmaclennan) [inline lib/polygon-feature.js]: This is cut & pasted from https://github.com/openstreetmap/iD/blob/9b7d79a3fec6f0eb5e292c3b89a2912fd8751197/modules/presets/index.js#L62-L104

**Comment by gmaclennan** [comment]
LGTM

**Pull request merged**